slack_sdk/web/slack_response.py

```diff
@@ -156,7 +158,7 @@ def __next__(self):
         else:
             raise StopIteration
 
-    def get(self, key, default=None):
+    def get(self, key: str, default: T = None) -> T:
```

Comment:
@eddyg I know that your current one also works but the "get" method can return either a value or None. So, if we want to make it more accurate, I think both the default value and returned value should be optional of something. What do you think?
```suggestion
    def get(self, key: str, default: Optional[T] = None) -> Optional[T]:
```

Reply:
This makes perfect sense! (I'm the first to admit I'm far from a Python typing expert 😰 so I very much appreciate you pointing this out!)

Code update incoming!

Reply:
Hmm... your comment made me try something.

With my change in place, the code below no longer generates any warnings _(notice the lack of a default value on the `get()` inside the `len()` now)_:

```python
    if api_response and api_response.get("ok") and len(api_response.get("messages")):
```

and I don't understand why. 🤔 I was expecting to see a warning like the one that caused me to investigate this issue in the first place.

Reply:
Hmm, with `Optional` added in, I'm back to a variation of my original problem:

```
Argument of type "list[Any] | None" cannot be assigned to parameter "__obj" of type "Sized" in function "len"
  Type "list[Any] | None" cannot be assigned to type "Sized"
    "__len__" is not present
```

This looks to be more complicated than I anticipated.

Reply:
What do you think of something like this approach? 🤔 

```python
    @overload
    def get(self, key: str, default: None = None) -> Optional[T]: ...
    @overload
    def get(self, key: str, default: T) -> T: ...
    def get(self, key, default):
```

Reply:
@eddyg Thanks for checking this. I also explored alternatives but could not find any. Considering that, it's fine to revert to your initial code. Sorry for bothering you but can you update this PR again?

Reply:
@eddyg Ah, overload is an interesting idea! but in this case, your initial code is not bad at all. Let's keep it simple

Reply:
The advantage of the `@overload` approach is that this ↓ code would generate the `__len__ is not present` warning, which the "simple" approach would not:

```python
    if api_response and api_response.get("ok") and len(api_response.get("messages"))
```

I'm happy to do whatever you prefer, including just abandoning this PR if it's more trouble than it's worth. (But it would be nice to get the typing issues with `.get()` resolved.)

Reply:
@eddyg Thanks, I see the benefit. As long as `@overload` code works and passes the CI builds, I am happy to merge it!

Reply:
I ran tests on this ↓ iteration of a fix (which addresses the `"Never" is not iterable` issue I noted above)

```
    @overload
    def get(self, key: str, default: None = None) -> Optional[Any]: ...
    @overload
    def get(self, key: str, default: T) -> T: ...
    def get(self, key, default=None):
```